src/cart/crt_rpc.h

```diff
@@ -67,6 +67,10 @@ struct crt_common_hdr {
 	d_rank_t	cch_src_rank;
 	/* tag to which rpc request was sent to */
 	uint32_t	cch_dst_tag;
+	/* flags */
+	/* indicates whether rpc originator intended to send on a primary ctx */
+	uint32_t	cch_src_is_primary : 1;
```

Comment:
is there any reason for not using bool here or to add that info as part of the flags ? also I'm wondering about potential struct padding here. If you intend to transmit that header over the wire, you might want to make sure that you use __attribute((packed))__  at the end of the struct to prevent the compiler from adding padding between the fields.

Reply:
I was under impression that we already do #pragma pack(4) or some such in every header, but perhaps it is somehow done through SCons/flags -- @jolivier23 do you know if/ow we force alignment/padding through build system?

Reply:
as for not using bool, i dont want to reserve whole uint32 for 1 flag, it is likely in the future we might transmit additional one, for example 0-15 id of the secondary provider once we support multiple secondary providers.  but yes, if we dont do auto-packing then i will add explicit padding